### DISCUSSION TITLE
Scraping metrics

### DISCUSSION THREAD

**Discussion by davisoares454** [kubernetes-sigs/kubebuilder#2536] [Support] [answered]
Hello, all. I'm trying to get the metrics exported to Prometheus. I read the documentation about metrics in the Kubebuilder book, created cluster roles, role binding, etc, but when going to Prometheus and specifying the TLS certificates in the Kubebuilder Operator Service Monitor because of Istio, it shows to me "unexpected EOF" in the Status/Targets screen.
I've tried using curl from Prometheus pod to the controller pod with Istio disabled: `curl https://10.4.109.58:8443/metrics -k`
But it returns "Unauthorized".

I'm in doubt because the controller logs also says that the metrics are export and listening to 127.0.0.1:8080, but the documentation says that the metrics are exposed by default in 8443 port.
What am I doing wrong? Thanks

**Comment by camilamacedo86** [comment]
Could you please raise an issue instead and add all details over what version of kb you used to scaffold and etc?

**Reply by davisoares454** [reply, marked as answer]
Got it solved. The main problem was Istio in Strict mode (mTLS enabled), so I had to change the kube-rbac-proxy arg from "--secure-listen-address=0.0.0.0:8443" to "--insecure-listen-address=0.0.0.0:8443", and adding the Istio certificates in the generated serviceMonitor with the fields in spec.endpoints.tlsConfig.

Thanks!!!

**Comment by camilamacedo86** [comment]
Closing this one since no interaction is required anymore.

**Comment by Charlotte-br560** [comment]
There might be an issue with your Istio configuration or authentication for accessing metrics. For a similar scenario, you might find it beneficial to check how [Crawlbase](https://crawlbase.com/) handles metrics export to Prometheus, especially in environments involving Istio. Their documentation or community discussions could provide insights or solutions to help troubleshoot and resolve the "unexpected EOF" and "Unauthorized" issues you're facing. Good luck with resolving the problem!
